File: backend/src/main/java/com/beauty/ecommerce/cart/application/service/CartService.java
package com.beauty.ecommerce.cart.application.service;

import com.beauty.ecommerce.cart.application.port.in.CartUseCase;
import com.beauty.ecommerce.cart.application.port.out.CartPort;
import com.beauty.ecommerce.cart.domain.entity.CartItem;
import com.beauty.ecommerce.common.application.service.ActivityLogService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class CartService implements CartUseCase {

    private final CartPort cartPort;
    private final ActivityLogService activityLogService;

    @Override
    public List<CartItem> getCart(String email) {
        return cartPort.findByUserEmail(email);
    }

    @Override
    public void addToCart(String email, Long productId, Integer quantity, String variantName) {
        cartPort.save(email, productId, quantity, variantName);
        activityLogService.logActivity(null, email, ActivityLogService.GROUP_SHOPPING, "ADD_TO_CART", "Thêm sản phẩm ID " + productId + (variantName != null ? " (" + variantName + ")" : "") + " vào giỏ hàng (Số lượng: " + quantity + ")");
    }

    @Override
    public void updateQuantity(String email, Long productId, Integer quantity, String variantName) {
        cartPort.updateQuantity(email, productId, quantity, variantName);
    }

    @Override
    public void removeFromCart(String email, Long productId, String variantName) {
        cartPort.delete(email, productId, variantName);
        activityLogService.logActivity(null, email, ActivityLogService.GROUP_SHOPPING, "REMOVE_FROM_CART", "Xóa sản phẩm ID " + productId + (variantName != null ? " (" + variantName + ")" : "") + " khỏi giỏ hàng");
    }

    @Override
    public void clearCart(String email) {
        cartPort.clearCart(email);
    }
}

File: frontend/src/api/cartService.ts
import axiosInstance from './axiosInstance';

export interface CartItemRequest {
  productId: number;
  quantity: number;
  variantName?: string | null;
}

export const cartService = {
  getCart: async () => {
    const response = await axiosInstance.get('/cart');
    return response.data;
  },
  addToCart: async (data: CartItemRequest) => {
    const response = await axiosInstance.post('/cart', data);
    return response.data;
  },
  updateQuantity: async (data: CartItemRequest) => {
    const response = await axiosInstance.put('/cart', data);
    return response.data;
  },
  removeFromCart: async (productId: number, variantName?: string | null) => {
    const response = await axiosInstance.delete(`/cart/${productId}`, {
      params: { variantName }
    });
    return response.data;
  }
};

export default cartService;
